### PR TITLE
Add code for saving model data to DB

### PR DESCRIPTION
**Pull request by laksh22**
# Description

Once the user has made the desired changes to the model in the frontend, the newest data about the model is stored in the Redux store. The user can then click the __Save Changes__ button to save the changes to the database.

This PR creates the API endpoint for saving the details about a new model as well as the changes to an existing model. The API is also integrated with the frontend.

> Note: The testing has been done using Postman. The integration with the frontend has been done but not tested since the frontend still used the old API and database for authentication, labels, etc. I will test the integration and make the necessary fixes once the frontend is connected to the new API.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Created 2 new models. Sent an edit request for one of the models.

**Test Configuration**:
New model request:
model1

New model response:
model2

Saved preprocessing steps JSON:
model6

Edit model request:
model3

Edit model response:
model4

Save changes button(yet to be tested):
model5

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by laksh22** [comment]
@ivantha Let me know if any changes need to be made 👍

File: labellab-flask/api/controllers/modelscontroller.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
+from flask.views import MethodView
+from flask import make_response, request, jsonify, current_app
+from flask_jwt_extended import jwt_required
+
+from api.models.Model import Model, optional_params
+from api.models.Projects import Project
+
+class Save(MethodView):
```
Comment (carpecodeum) [inline labellab-flask/api/controllers/modelscontroller.py:8]: Hey @laksh22 could you specify the model name more distinguishably?

File: labellab-flask/api/controllers/modelscontroller.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    """
+    This method removes the access token on logout and stores the revoked token.
+    """
+
+    def post(self):
```
Comment (carpecodeum) [inline labellab-flask/api/controllers/modelscontroller.py:13]: Please add the `@jwt_required` here you may refer to the logout classes for the boilerplate :)

File: labellab-flask/api/main.py [hunk 3]
```diff
@@ -36,16 +36,19 @@ def register_additional_extensions(app):
...
     return None
 
+
 def register_shellcontext(app):
     """Register shell context objects."""
 
     def shell_context():
         """Shell context objects."""
-        return {"db": db, "User": models.User}
+        return {"db": db, "User": models.User, "Model": models.model}
```
Comment (carpecodeum) [inline labellab-flask/api/main.py:52]: Hey, i dont think that the `models` Model requires a shell context so you may remove this :)

File: labellab-flask/api/controllers/modelscontroller.py (hunk 1 above)
Comment (laksh22) [reply labellab-flask/api/controllers/modelscontroller.py:8]: Would something like `MLModel` instead of `Model` be better?

Comment (carpecodeum) [reply labellab-flask/api/controllers/modelscontroller.py:8]: Yes sounds good!

**Pull request merged**